Meter.c

```diff
@@ -144,11 +148,97 @@ ListItem* Meter_toListItem(Meter* this, bool moving) {
    return li;
 }
 
+/* ---------- GraphData ---------- */
+
+static GraphData* GraphData_new(size_t nValues, size_t nItems, bool isPercentGraph) {
+   // colors[] is designed to be two-dimensional, but without a column of row
+   // pointers (unlike var[m][n] declaration).
+   // GraphMeterMode_draw() will print this table in 90-deg counter-clockwise
+   // rotated form.
+   unsigned int colorRowSize;
+   if (nItems <= 1) { // 1 or less item
+      colorRowSize = 1;
+   } else if (isPercentGraph) { // Percent graph: a linear row of color cells.
+      colorRowSize = GRAPH_HEIGHT;
+   } else { // Non-percentage & dynamic scale: a binary tree of cells.
+      colorRowSize = 2 * GRAPH_HEIGHT;
+   }
+   GraphData* data = xCalloc(1, sizeof(GraphData) +
+                                sizeof(double) * nValues +
+                                sizeof(double) * (nItems + 2) * 2 +
+                                sizeof(int) * (nValues * colorRowSize));
+   data->values = (double*)(data + 1);
+   // values[nValues + 1]
+   // It's intentional that values[nValues] and stack1[0] share the same cell;
+   // the cell's value will be always 0.0.
+   data->stack1 = (double*)(data->values + nValues);
+   // stack1[nItems + 2]
+   data->stack2 = (double*)(data->stack1 + (nItems + 2));
+   // stack2[nItems + 2]
+   data->colors = (int*)(data->stack2 + (nItems + 2));
+   // colors[nValues * colorRowSize]
```

Comment:
What do we gain from allocating this all in one block, instead of separating these allocations by struct field?